sync_auditlog/models/rule.py

```diff
@@ -0,0 +1,455 @@
+            # stored in the database only at the end of the transaction, but
+            # their values exist in cache.
+            new_values = {}
+            fields_list = rule_model.get_auditlog_fields(self)
+            for new_record in new_records.sudo():
+                new_values.setdefault(new_record.id, {})
+                for fname, field in new_record._fields.items():
+                    if fname not in fields_list:
+                        continue
+                    new_values[new_record.id][fname] = field.convert_to_read(
+                        new_record[fname], new_record
+                    )
+            context = dict(self._context)
+            context.update({"auditlog_is_capturing": True})
+            additional_log_values = {
+                "log_type": log_type,
+                "uuid": uuid.uuid4(),
+                "timestamp": datetime.today(),
+                "resource_ids": self.ids,
+                "raw_args_kwargs": vals_list,
+                "context": context,
+                "state": "captured",
+            }
+
+            rule_model.sudo().with_context(
+                black_list_fields=black_list_fields, white_list_fields=white_list_fields
+            ).create_logs(
+                self.env.uid,
+                self._name,
+                new_records.ids,
+                "create",
+                None,
+                new_values,
+                additional_log_values,
+            )
+            return new_records
+
+        @api.model_create_multi
+        @api.returns("self", lambda value: value.id)
+        def create_fast(self, vals_list, **kwargs):
+            self = self.with_context(auditlog_disabled=True)
+            rule_model = self.env["auditlog.rule"]
+            vals_list2 = copy.deepcopy(vals_list)
+            new_records = create_fast.origin(self, vals_list, **kwargs)
+            new_values = {}
+            for vals, new_record in zip(vals_list2, new_records):
+                new_values.setdefault(new_record.id, vals)
+            context = dict(self._context)
+            context.update({"auditlog_is_capturing": True})
+            additional_log_values = {
+                "log_type": log_type,
+                "uuid": uuid.uuid4(),
+                "timestamp": datetime.today(),
+                "resource_ids": self.ids,
+                "raw_args_kwargs": vals_list,
+                "context": context,
+                "state": "captured",
+            }
+
+            rule_model.sudo().with_context(
+                black_list_fields=black_list_fields, white_list_fields=white_list_fields
+            ).create_logs(
+                self.env.uid,
+                self._name,
+                new_records.ids,
+                "create",
+                None,
+                new_values,
+                additional_log_values,
+            )
+            return new_records
+
+        return create_full if self.log_type == "full" else create_fast
```

Comment:
create_full/create_fast are not needed for the sync log records.
We roll out our won log logic that does not look into new/old values, only to args/kwargs.